### PR TITLE
CAMEL-19470 camel-mongodb : add option fullDocument for changeStreams...

### PR DESCRIPTION
**Pull request by GarridoKevin**
… consumer.

# Description

Added fullDocument option to mongodb endpoint to attach the updated document to changeStreams event. Possible values are 'default' and 'updateLookup' in mongo version < 6.0.

# Target

- [x] I checked that the commit is targeting the correct branch (note that Camel 3 uses `camel-3.x`, whereas Camel 4 uses the `main` branch)

# Tracking
- [x] If this is a large change, bug fix, or code improvement, I checked there is a [JIRA issue](https://issues.apache.org/jira/browse/CAMEL) filed for the change (usually before you start working on it).

# Apache Camel coding standards and style

- [x] I checked that each commit in the pull request has a meaningful subject line and body.

- [x] I formatted the code using `mvn -Pformat,fastinstall install && mvn -Psourcecheck`

### CODE REVIEW

**Review by oscerd — APPROVED** [review]

**Comment by GarridoKevin** [comment]
I will push another commit later for fixes for changes with camel-4.x

**Comment by essobedo** [comment]
> I will push another commit later for fixes for changes with [camel-4](https://issues.apache.org/jira/browse/CAMEL-4).x

Just to be clear, If the target branch of your PR is `main` then the fix will go in Camel 4

**Comment by GarridoKevin** [comment]
> > I will push another commit later for fixes for changes with [camel-4](https://issues.apache.org/jira/browse/CAMEL-4).x
> 
> Just to be clear, If the target branch of your PR is `main` then the fix will go in Camel 4

Yes, there is another separate PR for camel-3x. I opened this one without checking for change in the new version. 
It should be good now.

**Review by essobedo — COMMENTED** [review]
Hi, thx for the PR, just some small remarks/comments.

File: components/camel-mongodb/src/main/java/org/apache/camel/component/mongodb/MongoDbEndpoint.java [hunk 1]
```diff
@@ -664,15 +665,15 @@ public void setStreamFilter(String streamFilter) {
...
     }
 
-    public String getFullDocument() {
+    public FullDocument getFullDocument() {
         return fullDocument;
     }
 
     /**
      * Specifies whether changeStream consumer include a copy of the full document when modified by update operations.
      * Possible values are default and updateLookup.
```
Comment (essobedo) [inline components/camel-mongodb/src/main/java/org/apache/camel/component/mongodb/MongoDbEndpoint.java]: Seems to be out dated with the latest changes

File: components/camel-mongodb/src/main/java/org/apache/camel/component/mongodb/MongoDbEndpoint.java [hunk 2]
```diff
@@ -580,9 +581,9 @@ public String getTailTrackIncreasingField() {
     public MongoDbTailTrackingConfig getTailTrackingConfig() {
         if (tailTrackingConfig == null) {
             tailTrackingConfig = new MongoDbTailTrackingConfig(
-                    persistentTailTracking, tailTrackIncreasingField, tailTrackDb == null ? database : tailTrackDb,
-                    tailTrackCollection,
-                    tailTrackField, getPersistentId());
+                    persistentTailTracking, tailTrackIncreasingField,
+                    tailTrackDb == null ? database : tailTrackDb, tailTrackCollection, tailTrackField,
+                    getPersistentId());
```
Comment (essobedo) [inline components/camel-mongodb/src/main/java/org/apache/camel/component/mongodb/MongoDbEndpoint.java:586]: I don't see any changes, only reformatting, so please revert

**Review by davsclaus — APPROVED** [review]

**Pull request merged**